PYME/Analysis/points/multiview.py

```diff
@@ -2,30 +2,43 @@
 import numpy as np
 from PYME.Analysis.points.DeClump import pyDeClump
 
-def coalesce_dict_sorted(inD, assigned, keys, weights_by_key):  # , notKosher=None):
+def coalesce_dict_sorted(inD, assigned, keys, weights_by_key, n_coalesced_key='n_coalesced'):
     """
-    Agregates clumps to a single event
-    Note that this will evaluate the lazy pipeline events and add them into the dict as an array, not a code
-    object.
-    Also note that copying a large dictionary can be rather slow, and a structured ndarray approach may be preferable.
-    DB - we should never have a 'large' dictionary (ie there will only ever be a handful of keys)
+    Aggregates clumps to a single event. Input dictionary should have every key sorted by the 'assigned' input. Output
+    dictionary will additionally have a key:value pair of the number of points used to produce each output point.
 
-    Args:
-        inD: input dictionary containing fit results
-        assigned: clump assignments to be coalesced
-        keys: list whose elements are strings corresponding to keys to be copied from the input to output dictionaries
-        weights_by_key: dictionary of weights.
+    Parameters
+    ----------
+        inD: dict
+            input dictionary containing fit results
+        assigned: 1darray, list
+            clump assignments to be coalesced
+        keys: list
+            list whose elements are strings corresponding to keys to be copied from the input to output dictionaries
+        weights_by_key: dict
+            dictionary of weights for each key as an array or weighting method ('min', 'sum', or 'mean').
+        n_coalesced_key: str
+            key to store the number of points coalesced to create each point in the output dictionary
 
-    Returns:
-        fres: output dictionary containing the coalesced results
+    Returns
+    -------
+        clumped: dict
+            output dictionary containing the coalesced results
+
+    Notes
+    -----
+    That this will evaluate the lazy pipeline events and add them into the dict as an array, not a code object.
 
     """
     from PYME.Analysis.points.DeClump import deClump
 
-    NClumps = int(np.max(assigned) + 1)  # len(np.unique(assigned))  #
-
     clumped = {}
 
+    NClumps = int(np.max(assigned) + 1)
+    # record number of points being coalesced for each clump
+    uni, counts = np.unique(assigned.astype(int), return_counts=True)
```

Comment:
Do we need to worry about performance issues here?

Reply:
'''
timeit.timeit('np.unique(test, return_counts=True)', setup='import numpy as np; test = np.random.randint(0, 100, int(1e6))', number=100) / 100
0.03026972654974088
'''
The 'right' thing to do would be piggy backing on the loop being run in deClump.findClumpsN, and have it return a counts-per group. As to where this fits in our/my priority queue, maybe we leave it for now and come back if it bothers us/others downstream?

Reply:
The query was because I thought it was already piggy-backed for the standard (not multi-view) declumping. It was actually more the np.repeat that I have performance worries about.

Agree it's not high priority.